optimuspy.py

```diff
@@ -164,49 +164,46 @@ def main(instance_name: str, cube_name: str, view_name: str, process_name: str,
+            storage_dimension_order = tm1.cubes.get_storage_dimension_order(cube_name=cube_name)
+            logging.info(f"Original dimension order for cube '{cube_name}' is: '{storage_dimension_order}'")
             displayed_dimension_order = tm1.cubes.get_dimension_names(cube_name=cube_name)
-            measure_dimension_only_numeric = is_dimension_only_numeric(tm1, original_dimension_order[-1])
+            measure_dimension_only_numeric = is_dimension_only_numeric(tm1, storage_dimension_order[-1])
 
             permutation_results = list()
             try:
 
                 original_order = OriginalOrderExecutor(
                     tm1, cube_name, [view_name], process_name, displayed_dimension_order, executions,
-                    measure_dimension_only_numeric, original_dimension_order)
+                    measure_dimension_only_numeric, storage_dimension_order)
                 permutation_results += original_order.execute(reset_counter=True)
 
                 main_executor = MainExecutor(
                     tm1, cube_name, [view_name], process_name, displayed_dimension_order, executions,
-                    measure_dimension_only_numeric, fast)
+                    measure_dimension_only_numeric, fast, dimensions_to_exclude)
                 permutation_results += main_executor.execute()
 
                 optimus_result = OptimusResult(cube_name, permutation_results)
 
                 best_permutation = optimus_result.best_result
                 logging.info(f"Completed analysis for cube '{cube_name}'")
                 if not best_permutation:
-                    tm1.cubes.update_storage_dimension_order(cube_name, original_dimension_order)
+                    tm1.cubes.update_storage_dimension_order(cube_name, storage_dimension_order)
                     logging.info(
                         f"No ideal dimension order found for cube '{cube_name}'."
                         f"Please pick manually based on csv and png results.")
-
                 else:
                     best_order = best_permutation.dimension_order
                     if update:
                         tm1.cubes.update_storage_dimension_order(cube_name, best_order)
                         logging.info(f"Updated dimension order for cube '{cube_name}' to {best_order}")
                     else:
                         logging.info(f"Best order for cube '{cube_name}': {best_order}")
-                        tm1.cubes.update_storage_dimension_order(cube_name, original_dimension_order)
+                        tm1.cubes.update_storage_dimension_order(cube_name, storage_dimension_order)
                         logging.info(
-                            f"Restored original dimension order for cube '{cube_name}' to {original_dimension_order}")
-
-            except:
-                logging.error("Fatal error", exc_info=True)
+                            f"Restored original dimension order for cube '{cube_name}' to {storage_dimension_order}")
```

Comment:
Ideally we drop the term "original dimension order" consistently in the project and substitute it with something everyone can agree with :)